src/components/hooks.js

```diff
@@ -184,18 +185,38 @@ export function useGasPrice(enabled = true) {
   const [loading, setLoading] = useState(true)
   const [price, setPrice] = useState({})
 
-  const gasApi = 'https://www.gasnow.org/api/v3/gas/price'
   useEffect(() => {
-    fetch(gasApi)
-      .then(res => {
-        return res.json()
-      })
-      .then(({ data }) => {
-        setPrice(data)
-        setLoading(false)
-      })
-      .catch(() => '') // ignore error
+    try {
+      const run = async () => {
+        const provider = await getProvider()
+        const blockDetails = await provider.getBlock('latest')
+        if (blockDetails.baseFeePerGas) {
+          const baseFeeWei = utils.formatUnits(
+            blockDetails.baseFeePerGas,
+            'wei'
+          )
+          const price = {
+            slow: baseFeeWei + 2 * Math.pow(10, 9),
+            fast: baseFeeWei * 1.5 + 2 * Math.pow(10, 9)
+          }
+          setPrice(price)
+          setLoading(false)
+        } else {
+          const gasApi = 'https://www.gasnow.org/api/v3/gas/price'
```

Comment:
I'm guessing this branch is designed to handle testnets that still haven't upgraded - but here it will use the gas price of mainnet. Can we just use the gasprice RPC in ethers instead?